contract-examples/contracts/ExampleDeployerList.sol

```diff
@@ -4,12 +4,145 @@ pragma solidity ^0.8.0;
 import "@openzeppelin/contracts/access/Ownable.sol";
 import "./IAllowList.sol";
 import "./AllowList.sol";
+import "ds-test/src/test.sol";
+
+address constant DEPLOYER_LIST = 0x0200000000000000000000000000000000000000;
+address constant OTHER_ADDRESS = 0x0Fa8EA536Be85F32724D57A37758761B86416123;
 
 // ExampleDeployerList shows how ContractDeployerAllowList precompile can be used in a smart contract
 // All methods of [allowList] can be directly called. There are example calls as tasks in hardhat.config.ts file.
 contract ExampleDeployerList is AllowList {
   // Precompiled Allow List Contract Address
-  address constant DEPLOYER_LIST = 0x0200000000000000000000000000000000000000;
-
   constructor() AllowList(DEPLOYER_LIST) {}
+
+  function deployContract() public {
+    new Example();
+  }
+}
+
+contract Example {}
```

Comment:
What is this for? could we add some comments?

Reply:
I feel like there's a bit of an impedance mismatch between what *I think* comments should be used for and how we treat them in `subnet-evm` in general. 

The only way you can tell what any `contract`, `struct`, or variables, etc is used for is to look at the call-site. 
Here in this case:
https://github.com/ava-labs/subnet-evm/blob/6bb15f7a6f60bbddf95546574b53a11d22d315da/contract-examples/contracts/ExampleDeployerList.sol#L18-L20

Then there are a couple of different places that `deployContract` is called:
https://github.com/ava-labs/subnet-evm/blob/6bb15f7a6f60bbddf95546574b53a11d22d315da/contract-examples/contracts/ExampleDeployerList.sol#L65-L73
and
https://github.com/ava-labs/subnet-evm/blob/6bb15f7a6f60bbddf95546574b53a11d22d315da/contract-examples/contracts/ExampleDeployerList.sol#L107-L125

The first call to `deployContract` tests whether or not `tx.origin` (the signer) can deploy a contract. This seems relatively clear to me, but I wrote it, so maybe it's not... but then I would think that you would ask me to clarify `test_noRoleCannotDeploy` instead of what `contarct Example {}` is. 

The second is making sure that we can deploy a contract through a different contract call.


So, just to be clear, I didn't add a big long comment here to try and make the case that the tests are crystal clear and couldn't use comments or better code to make them clearer, I just don't see how added a comment at this line helps clarify anything. If today I add a comment here explaining what `Example` is used for, then that changes, I likely wouldn't remember to go update the comment, and it would be misleading; it's a "nonlocal" comment.

Reply:
In general, non-obvious code and top level definitions should be accompanied by comments.

This is so the reviewers and maintainers can understand the code without seeking its callsites. It also helps confirm their understanding of the code after they read it, or allows them to skip over it and come back later if what they are looking for is not there.

You are correct in that this leads to some duplication and possibility of being outdated, however these comments are part of the code and we try to maintain their accuracy at a similar level to code.

For example, below we could have
```
// ExampleDeployerListTest defines transactions that are used to test the DeployerAllowList precompile by instantiating and calling the ExampleDeployerList and making assertions. The transactions are put together as steps of a complete test in contract_deployer_allow_list.ts.
```

Reply:
I don't want to keep going back and forth so I'll do it, but this violates "clean-code" principles.
https://medium.com/codex/clean-code-comments-833e11a706dc#e45c for reference. 

Using comments to understand code could lead to a false understanding if anything at the call site changes.